app/src/routes/project/TopBar/ModeToggle.tsx

```diff
@@ -1,12 +1,20 @@
 import { ToggleGroup, ToggleGroupItem } from '@/components/ui/toggle-group';
 import { EditorMode } from '@/lib/models';
 import { observer } from 'mobx-react-lite';
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 import { useEditorEngine } from '..';
 
 const ModeToggle = observer(() => {
     const editorEngine = useEditorEngine();
-    const [mode, setMode] = useState<EditorMode>(editorEngine.mode);
+    const [mode, setMode] = useState<EditorMode>(makeDesignMode(editorEngine.mode));
+
+    useEffect(() => {
+        setMode(makeDesignMode(editorEngine.mode));
```

Comment:
Update this to handle pan mode changing from interact mode. Now it will update the top bar as well.